readthedocs/builds/filters.py

```diff
@@ -64,7 +64,7 @@ def get_state(self, queryset, _, value):
 
     def get_type(self, queryset, _, value):
         if value == self.TYPE_NORMAL:
-            queryset = queryset.exclude(type=EXTERNAL)
+            queryset = queryset.exclude(version__type=EXTERNAL)
         elif value == self.TYPE_EXTERNAL:
-            queryset = queryset.filter(type=EXTERNAL)
+            queryset = queryset.filter(version__type=EXTERNAL)
```

Comment:
We should use `field_name="version__type"` in the definition of the field, similar to what we are doing for `version=` field. This way we keep consistency and we know we are asking for the Version type instead of the Build type and avoid this confusion in the future.

Reply:
Oops, actually this is getting a bit gnarlier than I expected, `self.state` is not an attribute on the `FilterSet` instance as I think fields are managed through a metaclass.

Reply:
@humitos Okay, I did find a fix, though I can't tell if it's an improvement or not -- it is a bit harder to follow:

https://github.com/readthedocs/readthedocs.org/commit/5749f2c52

Reply:
Ah okay, that's a little different then. The attribute `field_name` is the name of the _model field_ to use for the filter's query. The name of the filter, exposed as a URL param, is the name of the filter attribute on the filterset.

To accomplish what you are looking for, you need this instead: d44628373